numba/testing/main.py

```diff
@@ -84,19 +85,36 @@ def cuda_sensitive_mtime(x):
 
     return key
 
+
 def parse_slice(useslice):
-    """Parses the argument string "useslice" as the arguments to the `slice()`
-    constructor and returns a slice object that's been instantiated with those
-    arguments. i.e. input useslice="1,20,2" leads to output `slice(1, 20, 2)`.
+    """Parses the argument string "useslice" as a shard index and number and
+    returns a function that filters on those arguments. i.e. input
+    useslice="1:3" leads to output something like `lambda x: zlib.crc32(x) % 3
+    == 1`.
     """
+    if callable(useslice):
+        return useslice
+    if not useslice:
+        return lambda x: True
     try:
-        l = {}
-        exec("sl = slice(%s)" % useslice, l)
-        return l['sl']
+        (index, count) = useslice.split(":")
+        index = int(index)
+        count = int(count)
     except Exception:
-        msg = ("Expected arguments consumable by 'slice' to follow "
-                "option `-j`, found '%s'" % useslice)
+        msg = (
+                    "Expected arguments shard index and count to follow "
+                    "option `-j`, found '%s'" % useslice)
```

Comment:
This doesn't explain how to specify them correctly? It's expecting syntax `-j index:count` ?